games-arcade/freegish/patches/freegish-1.0.patchset

```diff
@@ -0,0 +1,138 @@
+   SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER,1);
+-
+-
+-  Uint32 videoModeFlags = SDL_OPENGL;
++  
++  
++  videoModeFlags = SDL_OPENGL;
+   if (windowinfo.fullscreen)
+     videoModeFlags |= SDL_FULLSCREEN;
+-  SDL_Surface *surface = SDL_SetVideoMode(windowinfo.resolutionx,windowinfo.resolutiony,windowinfo.bitsperpixel,videoModeFlags);
++  *surface = *SDL_SetVideoMode(windowinfo.resolutionx,windowinfo.resolutiony,windowinfo.bitsperpixel,videoModeFlags);
+   if (!surface)
+     {
+     fprintf(stderr, "Failed to initialize video:\n%s\n",SDL_GetError());
+diff --git a/src/video/opengl.h b/src/video/opengl.h
+index 77ce8bc..aa030b0 100644
+--- a/src/video/opengl.h
++++ b/src/video/opengl.h
+@@ -37,7 +37,7 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ #include "../video/glext.h"
+ #endif
+ 
+-#if defined(LINUX) || defined(__FreeBSD_kernel__) || defined(__GNU__)
++#if defined(LINUX) || defined(__FreeBSD_kernel__) || defined(__GNU__) || defined(__HAIKU__)
+ #include <GL/gl.h>
+ #include <GL/glext.h>
+ #endif
+diff --git a/src/video/texture.c b/src/video/texture.c
+index 69ecae4..bc86e7f 100644
+--- a/src/video/texture.c
++++ b/src/video/texture.c
+@@ -52,6 +52,11 @@ int loadtexturepng(const char *filename, unsigned int **rgba, int *width, int *h
+ {
+ 	FILE *fp;
+ 	int load_status;
++	png_uint_32 texwidth;
++	png_uint_32 texheight;
++	int bit_depth;
++    int color_type;
++    int interlace_type;
+ 	
+ 	if((fp = fopen(filename, "rb")) == NULL)
+ 	{
+@@ -88,11 +93,11 @@ int loadtexturepng(const char *filename, unsigned int **rgba, int *width, int *h
+ 						png_init_io(png_ptr, fp);
+ 						png_set_sig_bytes(png_ptr, 8);
+ 						png_read_info(png_ptr, info_ptr);
+-						png_uint_32 texwidth = 0;
+-						png_uint_32 texheight = 0;
+-						int bit_depth = 0;
+-                        			int color_type = 0;
+-                        			int interlace_type = 0;
++						texwidth = 0;
++						texheight = 0;
++						bit_depth = 0;
++                        			color_type = 0;
++                        			interlace_type = 0;
+ 			                        png_get_IHDR(png_ptr, info_ptr, &texwidth, &texheight, &bit_depth, &color_type, 0, 0, 0);
+ 
+ 						/* expand paletted colors into true rgb */
+-- 
+2.2.2
+
+
+From a5d9b7e14969d7e03ea9d6718b937667c3159273 Mon Sep 17 00:00:00 2001
+From: Damien Lee <hometue.lee@gmail.com>
+Date: Tue, 5 Jan 2016 19:15:07 +0000
+Subject: Removed hardcoding of data dir
+
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 3841394..618d158 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -14,7 +14,7 @@ endif(NOT ${INSTALL_FHS})
+ # Uncomment this to hardcode the data path. On the command line, you
+ # would actually give -DDATAPATH='"/usr/share/games/gish"' or similar
+ # but CMake does the extra quoting for us.
+-add_definitions(-DDATAPATH="/boot/home/config/non-packaged/bin/")
++#add_definitions(-DDATAPATH="/boot/home/config/non-packaged/bin/")
```

Comment:
That was from the original file. I uncommented it but it didn't work, so I commented it back.